lib/config_cat/config/percentage_rule.ex

```diff
@@ -0,0 +1,32 @@
+defmodule ConfigCat.Config.PercentageRule do
+  @moduledoc false
+  alias ConfigCat.Config
+
+  @type t :: %{String.t() => term()}
+
+  @percentage "p"
+  @value "v"
+  @variation_id "i"
+
+  @spec percentage(t()) :: non_neg_integer()
+  def percentage(rule) do
+    Map.get(rule, @percentage, 0)
+  end
+
+  @spec value(t()) :: Config.value()
+  def value(rule) do
+    Map.get(rule, @value)
+  end
+
+  @spec variation_id(t()) :: Config.variation_id() | nil
```

Comment:
@kp-cat Can I safely assume that some of these config fields in this file and others will always be present? I'd like to be able to use `Map.fetch!` which will raise an error on a missing field rather than silently returning `nil`.

If I can make these assumptions, which fields are guaranteed to be present?